index.html

```diff
@@ -359,30 +359,80 @@
-          lifetime.
-        </p>
-        <p>
-          When a <a>push subscription</a> is <dfn data-lt=
-          "deactivate">deactivated</dfn>, both the <a>user agent</a> and the <a>push
-          service</a> MUST delete any stored copies of its details. Subsequent <a>push messages</a>
-          for this <a>push subscription</a> MUST NOT be delivered.
-        </p>
-        <p>
-          A <a>push subscription</a> is <a>deactivated</a> when its associated <a>service worker
-          registration</a> is unregistered, though a <a>push subscription</a> MAY be
-          <a>deactivated</a> earlier. A <a>push subscription</a> is removed when the <a>clear
-          registration</a> algorithm is run for the <a>service worker registration</a>.
-        </p>
-        <p>
           A <a>push subscription</a> has an associated <dfn data-lt=
           "push endpoints">push endpoint</dfn>. It MUST be the absolute URL exposed
           by the <a>push service</a> where the <a>application server</a> can send <a>push
           messages</a> to. A <a>push endpoint</a> MUST uniquely identify the <a>push
           subscription</a>.
         </p>
+        <p>
+          A <a>push subscription</a> has internal slots for a P-256 <a>ECDH</a> key pair and an
+          authentication secret in accordance with [[!WEBPUSH-ENCRYPTION]]. These slots MUST be
+          populated when creating the <a>push subscription</a>.
+        </p>
+        <p>
+          If the <a>user agent</a> has to change the keys for any reason,
+          it MUST <a>fire the pushsubscriptionchange event</a> with the
+          <a>service worker registration</a> associated with the <a>push subscription</a> as
+          <var>registration</var>, a <a>PushSubscription</a> instance representing the
+          <a>push subscription</a> having the old keys as <var>oldSubscription</var> and a
+          <a>PushSubscription</a> instance representing the <a>push subscription</a> having the new
+          keys as <var>newSubscription</var>.
+        </p>
+        <section>
+          <h2>
+            Subscription refreshes
+          </h2>
+          <p>
+            A <a>user agent</a> or <a>push service</a> MAY choose to <dfn>refresh</dfn> a
+            <a>push subscription</a> at any time, for example because it has reached a certain age.
+          </p>
+          <p>
+            When this happens, the <a>user agent</a> MUST create a new <a>push subscription</a>
+            with the <a>push service</a> on behalf of the application, using the
+            <a>PushSubscriptionOptions</a> that were provided for creating the current
+            <a>push subscription</a>. The new <a>push subscription</a> MUST have a key pair that's
+            different from the original subscription.
+          </p>
+          <p>
+            When successful, <a>user agent</a> then MUST <a>fire the pushsubscriptionchange event</a>
```

Comment:
I added the following paragraph:

> If the user agent is not able to refresh the push subscription, it should periodically retry the refresh. When the push subscription can no longer be used, for example because it has expired, the user agent must fire the pushsubscriptionchange event with the service worker registration associated with the push subscription as registration, a PushSubscription instance representing the deactivating push subscription as oldSubscription and null as the newSubscription.

I like Kit's suggestion of setting `newSubscription` to `null` since it gives the developer _something_.